frontend/lib/shared/payment_details_page.dart

```diff
@@ -0,0 +1,191 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+import 'package:flutter_starter/l10n/app_localizations.dart';
+import 'package:flutter_starter/shared/grid.dart';
+import 'package:flutter_starter/shared/json_schema_form.dart';
+import 'package:flutter_starter/shared/payment_method.dart';
+import 'package:flutter_starter/shared/search_payment_methods_page.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+
+class PaymentDetailsPage extends HookConsumerWidget {
+  const PaymentDetailsPage({super.key});
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final paymentMethods = ref.watch(paymentMethodProvider);
+
+    final selectedPaymentMethod = useState<PaymentMethod>(paymentMethods.first);
+    final selectedPaymentType =
```

Comment:
> we probably omit a state and write a transform method for this instead if it's only for getting the bank name. then we only need to track selectedPaymentMethod.

`selectedPaymentType` is also being used for the `selected` property in `SegmentedButton`

Reply:
> also do we want to default to a payment method? should we present an empty state and allow the user to choose from the search page?

think defaulting to a payment method makes sense here as a lot of offerings we've been working with only have one payin/payout method

Reply:
> > selectedPaymentType is also being used for the selected property in SegmentedButton
> 
> yee! we can do `SegmentedButton<PaymentMethod>` to avoid that - commented below but the editor somehow cut off `<PaymentMethod>` 🥴

hmm wouldn't that mean there would be `n` segments if the list of payment methods has a length of `n`?

Reply:
yep, that spec change is already being tracked in https://github.com/TBD54566975/tbdex/issues/233